server/plugin.go

```diff
@@ -394,3 +397,112 @@ func (p *Plugin) dm(userID string, message string) error {
 	}
 	return nil
 }
+
+func (p *Plugin) setUserStatus(userID string, meetingID int, meetingEnd bool) error {
+	statusChangePrefkey := fmt.Sprintf("%v_%v", changeStatusKey, userID)
+	changeStatus, err := p.API.KVGet(statusChangePrefkey)
+	if err != nil {
+		p.API.LogDebug("Could not get old status from KVStore", "err", err.Error())
+		return err
+	}
+
+	if string(changeStatus) != yes {
+		return nil
+	}
+
+	statusKey := fmt.Sprintf("%v%v", oldStatusKey, meetingID)
+	if meetingEnd {
+		b, appErr := p.API.KVGet(statusKey)
+		if appErr != nil {
+			p.API.LogDebug("Could not get old status from KVStore", "err", appErr.Error())
+			return appErr
+		}
+
+		newStatus := string(b)
+		if newStatus == "" {
+			newStatus = model.STATUS_ONLINE
+		}
+
+		_, appErr = p.API.UpdateUserStatus(userID, newStatus)
+		if appErr != nil {
+			p.API.LogDebug("Could not get update status", "err", appErr.Error())
+			return appErr
+		}
+
+		return nil
+	}
+
+	currentStatus, appErr := p.API.GetUserStatus(userID)
+	if appErr != nil {
+		p.API.LogDebug("Failed to update user status", "err", appErr)
+		return appErr
+	}
+
+	oldStatus := ""
+	if currentStatus.Manual {
+		oldStatus = currentStatus.Status
+	}
+
+	appErr = p.API.KVSetWithExpiry(statusKey, []byte(oldStatus), meetingPostIDTTL)
+	if appErr != nil {
+		p.API.LogDebug("failed to store old status", "err", appErr)
+		return appErr
+	}
+
+	_, appErr = p.API.UpdateUserStatus(userID, model.STATUS_DND)
+	if appErr != nil {
+		p.API.LogDebug("Failed to update user status", "err", appErr)
+		return appErr
+	}
+
+	return nil
+}
+
+func (p *Plugin) sendStatusChangeAttachment(userID, botUserID string, meetingID int) error {
+	url := pluginURLPath + postActionPath
+	actionYes := &model.PostAction{
+		Name: "Yes",
+		Integration: &model.PostActionIntegration{
+			URL: url,
+			Context: map[string]interface{}{
+				"accept":    true,
+				"meetingId": meetingID,
+			},
+		},
+	}
+
+	actionNo := &model.PostAction{
+		Name: "No",
+		Integration: &model.PostActionIntegration{
+			URL: url,
+			Context: map[string]interface{}{
+				"accept":    false,
+				"meetingId": meetingID,
+			},
+		},
+	}
+
+	sa := &model.SlackAttachment{
+		Title:   "Status change",
+		Text:    "Allow Zoom plugin to automatically change status",
+		Actions: []*model.PostAction{actionYes, actionNo},
+	}
+
+	attachmentPost := model.Post{}
+	model.ParseSlackAttachment(&attachmentPost, []*model.SlackAttachment{sa})
+	directChannel, appErr := p.API.GetDirectChannel(userID, botUserID)
```

Comment:
I feel it makes more sense to send this as an ephemeral on the same channel the meeting is created.

Reply:
I had some issues getting it to work as an ephemeral message. The buttons were not responsive.

Reply:
Could you share more information about the problem there? There might be a bug. Is there anything relevant in the logs / browser console?

Reply:
It is able to update the settings but once I choose the option, the post does not collapse or disappear. Is that the expected behaviour?

Reply:
Shouldn't be. Could you implement it with ephemeral posts (maybe in a branch if you prefer) so I can check it and see what is going on?

Reply:
@larkox I have pushed the changes to a new branch [https://github.com/cvhariharan/mattermost-plugin-zoom/tree/emphemeral-post](https://github.com/cvhariharan/mattermost-plugin-zoom/tree/emphemeral-post)

Reply:
@larkox any updates on this?

Reply:
I will try to give you an answer between today and tomorrow. Thank you for your patience!

Reply:
There seems to be a bug there indeed. Solve it your way, and we will consider changing it in the future is the bug gets solved.